### PR TITLE
Add clang-tidy checks for Firestore

### PR DESCRIPTION
**Pull request by wilhuff**
For now this is only really visible via CLion, when you disable the "use IDE configure" for the clang-tidy inspection.

I've run out of time trying to get CMake to also run these checks though this is something I also want to do.

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/remote/serializer.h [hunk 1]
```diff
@@ -72,7 +72,7 @@ class Serializer {
   // TODO(rsgowman): If we never support any output except to a vector, it may
   // make sense to have Serializer own the vector and provide an accessor rather
   // than asking the user to create it first.
-  static util::Status EncodeFieldValue(
```
Comment (zxu123) [inline Firestore/core/src/firebase/firestore/remote/serializer.h]: could you elaborate why remove `static` here?

File: Firestore/core/src/firebase/firestore/util/comparator_holder.h [hunk 2]
```diff
@@ -47,7 +47,7 @@ class ComparatorHolder {
 template <typename C>
 class ComparatorHolder<C, true> : private C {
  protected:
-  explicit ComparatorHolder(const C&) noexcept {
+  explicit ComparatorHolder(const C& /* comparator */) noexcept {
```
Comment (zxu123) [inline Firestore/core/src/firebase/firestore/util/comparator_holder.h]: Maybe put the parameter name outside of comment like
`explicit ComparatorHolder(const C& comparator) noexcept {`

File: Firestore/core/src/firebase/firestore/remote/serializer.h (hunk 1 above)
Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/remote/serializer.h]: The tests instantiate the serializer and then access these static methods through the instance, which causes a warning under clang-tidy.

In the fullness of time the serializer will need to be stateful (for example, to encode a DocumentKey you need to know the current DatabaseId) so fixing those warnings by removing the static seemed like the better way.

File: Firestore/core/src/firebase/firestore/util/comparator_holder.h (hunk 2 above)
Comment (wilhuff) [reply Firestore/core/src/firebase/firestore/util/comparator_holder.h]: That causes a different warning about an unused parameter. This way both the compiler and the linter are happy.

I could also do
```c++
explicit ComparatorHolder(const C& comparator) noexcept {
  (void)comparator;
}
```
But considered this to be inferior. I have no strong preference though.

**Pull request merged**